origin/globus_client.go

```diff
@@ -560,75 +584,98 @@ func handleGlobusAuth(ctx *gin.Context) {
 	ctx.Redirect(http.StatusTemporaryRedirect, redirectUrl)
 }
 
-// Persist the access token on the disk
-func persistAccessToken(collectionID string, token *oauth2.Token) error {
+// Persist a Globus access token on the disk
+func persistToken(collectionID string, token *oauth2.Token, tokenType TokenType) (string, error) {
 	uid, err := config.GetDaemonUID()
 	if err != nil {
-		return errors.Wrap(err, "failed to persist Globus access token on disk: failed to get uid")
+		return "", errors.Wrap(err, fmt.Sprintf("failed to persist Globus %s access token on disk: failed to get uid", tokenType))
 	}
 
 	gid, err := config.GetDaemonGID()
 	if err != nil {
-		return errors.Wrap(err, "failed to persist Globus access token on disk: failed to get gid")
+		return "", errors.Wrap(err, fmt.Sprintf("failed to persist Globus %s access token on disk: failed to get gid", tokenType))
 	}
 	globusFdr := param.Origin_GlobusConfigLocation.GetString()
 	tokBase := filepath.Join(globusFdr, "tokens")
 	if filepath.Clean(tokBase) == "" {
-		return fmt.Errorf("failed to update Globus token: Origin.GlobusTokenLocation is not a valid path: %s", tokBase)
+		return "", fmt.Errorf("failed to update Globus %s token: Origin.GlobusTokenLocation is not a valid path: %s", tokenType, tokBase)
 	}
-	tokFileName := filepath.Join(tokBase, collectionID+GlobusTokenFileExt)
-	tmpTokFile, err := os.CreateTemp(tokBase, collectionID+GlobusTokenFileExt)
+
+	var filename string
+	switch tokenType {
+	case TokenTypeCollection:
+		filename = collectionID + GlobusTokenFileExt
+	case TokenTypeTransfer:
+		filename = collectionID + GlobusTransferTokenFileExt
+	default:
+		return "", fmt.Errorf("unknown token type: %s", tokenType)
+	}
+
+	tokFileName := filepath.Join(tokBase, filename)
+	tmpTokFile, err := os.CreateTemp(tokBase, filename)
 	if err != nil {
-		return errors.Wrap(err, "failed to update Globus token: unable to create a temporary Globus token file")
+		return "", errors.Wrap(err, fmt.Sprintf("failed to update Globus %s token: unable to create a temporary Globus %s token file", tokenType, tokenType))
```

Comment:
To make this a bit more consistent with the rest of the codebase, can you use `errors.Wrapf` instead of using an `fmt.Sprintf` in the wrapped error, e.g  `errors.Wrapf(err, "failed to ... %s", var)`?

Same comment for the other errors immediately below -- although one already uses `errors.Wrapf`.